### PR TITLE
Do not use username_algo for mozillians.org profiles.

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

File: moderator/moderate/auth.py [hunk 1]
```diff
@@ -9,25 +9,48 @@
...
-                                             settings.MOZILLIANS_API_KEY)
-
         email = claims.get('email')
         try:
-            mozillian_user = mozillians_client.lookup_user({'email': email})
+            self.mozillian_user = self.mozillians_client.lookup_user({'email': email})
         except (BadStatusCode, ResourceDoesNotExist):
+            pass
+
+        if not self.mozillian_user:
```
Comment (johngian) [inline moderator/moderate/auth.py]: Why do you need this change? I would prefer to ``return None`` in except case.

File: moderator/moderate/auth.py [hunk 2]
```diff
@@ -9,25 +9,48 @@
...
             users = User.objects.filter(email__in=user_emails)
             if users:
                 return users[0]
             return super(ModeratorAuthBackend, self).create_user(claims, **kwargs)
         return None
+
+    def authenticate(self, **kwargs):
+        """Override authenticate method of the OIDC lib."""
+        user = super(ModeratorAuthBackend, self).authenticate(**kwargs)
+        profile = user.userprofile
```
Comment (johngian) [inline moderator/moderate/auth.py]: What happens if authenticate returns ``None``?

File: moderator/moderate/auth.py [hunk 3]
```diff
@@ -9,25 +9,48 @@
...
+        if not self.mozillian_user:
+            # existing user, let's fetch mozillians data
+            self.mozillian_user = self.mozillians_client.lookup_user({'email': user.email})
+
+        if not profile.username or not profile.username == self.mozillian_user['username']:
```
Comment (johngian) [inline moderator/moderate/auth.py]: Isn't this check enough here:
```
if not profile.username == self.mozillians_user['username']:
```

**Comment by akatsoulas** [comment]
Thanks @johngian 🍰

**Pull request merged**